airflow/jobs/local_task_job.py

```diff
@@ -73,15 +73,9 @@ def _execute(self):
         # pylint: disable=unused-argument
         def signal_handler(signum, frame):
             """Setting kill signal handler"""
-            self.log.error("Received SIGTERM. Terminating subprocesses")
-            self.on_kill()
-            self.task_instance.refresh_from_db()
-            if self.task_instance.state not in State.finished:
-                self.task_instance.set_state(State.FAILED)
-            self.task_instance._run_finished_callback(  # pylint: disable=protected-access
-                error="task received sigterm"
-            )
-            raise AirflowException("LocalTaskJob received SIGTERM signal")
+            self.log.error("LocalTaskJob received SIGTERM signal")
+            # Call the handle_task_exit method with SIGTERM exit code 143:
+            self.handle_task_exit(143)
```

Comment:
```suggestion
            self.handle_task_exit(128 + signum)
```

would be better I think (this is the default behaviour